### PR TITLE
Clam 2585 ole check encrypted 3

### PR DESCRIPTION
**Pull request by ragusaa**

### CODE REVIEW

**Comment by micahsnyder** [comment]
Replaced by https://github.com/Cisco-Talos/clamav/pull/1279

**Comment by micahsnyder** [comment]
My mistake, this one replaces the other.

File: libclamav/ole2_extract.c [hunk 1]
```diff
@@ -2635,13 +2937,33 @@ cl_error_t cli_ole2_extract(const char *dirname, cli_ctx *ctx, struct uniq **fil
...
         } else {
-            ret = ole2_walk_property_tree(&hdr, NULL, 0, handler_otf, 0, &file_count, ctx, &scansize2, NULL);
+            ret = ole2_walk_property_tree(&hdr, NULL, 0, handler_otf, 0, &file_count, ctx, &scansize2, NULL, &encryption_status);
+        }
+    }
+
+
+    if (SCAN_COLLECT_METADATA && (ctx->wrkproperty != NULL)) {
+        if (encryption_status.encrypted){
+            if (encryption_status.encryption_type){
```
Comment (micahsnyder) [inline libclamav/ole2_extract.c:2949]: Minor formatting issues:
- extra new line 
- missing space between `){`

File: libclamav/ole2_extract.c [hunk 2]
```diff
@@ -586,6 +610,259 @@ static cl_error_t handler_enum(ole2_header_t *hdr, property_t *prop, const char
...
+
+const char *const RC4_ENCRYPTION    = "RC4";
+const char *const XOR_OBFUSCATION   = "XORObfuscation";
+const char *const AES128_ENCRYPTION = "AES128";
+const char *const AES192_ENCRYPTION = "AES192";
+const char *const AES256_ENCRYPTION = "AES256";
+const char * const VELVET_SWEATSHOP_ENCRYPTION = "VelvetSweatshop";
+                const char * const GENERIC_ENCRYPTED = "ENCRYPTION_TYPE_UNKNOWN";
+
+        const char * const OLE2_HEURISTIC_ENCRYPTED_WARNING = "Heuristics.Encrypted.OLE2";
```
Comment (micahsnyder) [inline libclamav/ole2_extract.c:707]: Formatting got all funky here.

**Review by micahsnyder — CHANGES_REQUESTED** [review]
Other the formatting, I don't have any other concerns.

File: libclamav/ole2_extract.c (hunk 2 above)
Comment (ragusaa) [reply libclamav/ole2_extract.c]: Ran clam-format.

File: libclamav/ole2_extract.c (hunk 1 above)
Comment (ragusaa) [reply libclamav/ole2_extract.c]: Ran clam-format.

**Comment by ragusaa** [comment]
I ran clam-format to clean up the formatting, so I think this one is good to go.

**Comment by micahsnyder** [comment]
Just rebased with upstream main to get the CI fixes and other clang-format fixes.

**Review by micahsnyder — APPROVED** [review]

**Comment by micahsnyder** [comment]
Rebased again

**Pull request merged**